app/views/layouts/application.html.erb

```diff
@@ -48,7 +48,7 @@
     <% cache(release_adjusted_cache_key("top-html-and-config--#{user_signed_in?}")) do %>
       <body
         data-user-status="<%= user_logged_in_status %>"
-        class="<%= Settings::UserExperience.default_font.tr("_", "-") %>-article-body default-header"
+        class="<%= Settings::UserExperience.default_font.tr("_", "-") %>-article-body <%= creator_onboarding_mode ? "no-header" : "default-header" %>"
```

Comment:
If I am logged out and I manually change the return of  `creator_onboarding_mode` from true to false, this conditional ( <%= creator_onboarding_mode ? "no-header" : "default-header" %>` ) seems to work. 

However, I would have expected this **not to work** because  the only time the cache is updated is when the user_signed_in. I would have expected that I'd need to add `creator_onboarding_mode` to the cache key. This feels like a silly question but I guess I'm confused on why its working, any insights @benhalpern or @atsmith813 ?

Reply:
In addition, are there any other techniques that I should consider. I'm thinking of moving this into a [separate layout ](https://forem-team.slack.com/archives/C01S4QZUDNC/p1632137443006500) instead of littering the code with conditionals, but I'm still figuring out how that would work with caching.